policy-controller/k8s/index/src/tests/http_routes.rs

```diff
@@ -76,6 +79,118 @@ fn route_attaches_to_server() {
     );
 }
 
+#[test]
+fn routes_created_for_probes() {
+    let policy = DefaultPolicy::Allow {
+        authenticated_only: false,
+        cluster_only: true,
+    };
+    let probe_networks = vec!["10.0.0.1/24".parse().unwrap()];
+    let test = TestConfig::from_default_policy_with_probes(policy, probe_networks);
+
+    // Create a pod.
+    let container = k8s::Container {
+        liveness_probe: Some(k8s::Probe {
+            http_get: Some(k8s::HTTPGetAction {
+                path: Some("/liveness-container-1".to_string()),
+                port: k8s::IntOrString::Int(5432),
+                ..Default::default()
+            }),
+            ..Default::default()
+        }),
+        readiness_probe: Some(k8s::Probe {
+            http_get: Some(k8s::HTTPGetAction {
+                path: Some("/ready-container-1".to_string()),
+                port: k8s::IntOrString::Int(5432),
+                ..Default::default()
+            }),
+            ..Default::default()
+        }),
+        ..Default::default()
+    };
+    let mut pod = mk_pod_with_containers("ns-0", "pod-0", Some(container));
+    pod.labels_mut()
+        .insert("app".to_string(), "app-0".to_string());
+    test.index.write().apply(pod);
+
+    let mut rx = test
+        .index
+        .write()
+        .pod_server_rx("ns-0", "pod-0", 5432.try_into().unwrap())
+        .expect("pod-0.ns-0 should exist");
+
+    let mut expected_authorizations = HashMap::default();
+    expected_authorizations.insert(
+        AuthorizationRef::Default("probe"),
+        ClientAuthorization {
+            networks: vec!["10.0.0.1/24".parse::<IpNet>().unwrap().into()],
+            authentication: ClientAuthentication::Unauthenticated,
+        },
+    );
+    let liveness_match = HttpRouteMatch {
+        path: Some(PathMatch::Exact("/liveness-container-1".to_string())),
+        headers: vec![],
+        query_params: vec![],
+        method: Some(Method::GET),
+    };
+    let ready_match = HttpRouteMatch {
+        path: Some(PathMatch::Exact("/ready-container-1".to_string())),
+        headers: vec![],
+        query_params: vec![],
+        method: Some(Method::GET),
+    };
+
+    // No Server is configured for the port, so expect the probe paths to be
+    // authorized.
+    let update = rx.borrow_and_update();
+    let probes = update
+        .http_routes
+        .get(&InboundHttpRouteRef::Default("probe"))
+        .unwrap();
+    let probes_rules = probes.rules.get(0).unwrap();
+    assert!(probes_rules.matches.contains(&liveness_match));
+    assert!(probes_rules.matches.contains(&ready_match));
```

Comment:
nit, not a big deal: since these are just `assert!`s (and not e.g. `assert_eq!`s), they won't print out what `probes_rules` actually looks like if they fail. it _might_ be worth adding something like
```suggestion
    assert!(probes_rules.matches.contains(&liveness_match), "matches: {:#?}", probes_rules.matches);
    assert!(probes_rules.matches.contains(&ready_match), "matches: {:#?}", probes_rules.matches);
```